.github/workflows/ci-aot.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish AOTCompatibility testApp
```

Comment:
How does this action get hooked up into the CI system? Do we have to wait for it to merge before we can run it? Or is it already running on this PR?

Reply:
It's already running and can be seen from the "Checks" tab
 
![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/17327289/7b05c526-e273-4b39-bbdd-42e60e871cfa)

Reply:
It's already running with this PR: https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/5719889408/job/15498727666?pr=4717
This workflow will be triggered automatically on push: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on

https://github.com/open-telemetry/opentelemetry-dotnet/pull/4717/files#diff-65a128c6ce6f6956d1ff4c87ff6d27540e6faa4bcd3a2e59343a6947d8e83c15:~:text=on%3A,push%3A